nodes/src/nodes/library/ssrf_protection.py

```
@@ -0,0 +1,282 @@
+        pass  # not an IP literal — resolve via DNS
+
+    try:
+        addrinfos = socket.getaddrinfo(hostname, port, proto=socket.IPPROTO_TCP)
+    except socket.gaierror as exc:
+        raise SSRFError(f'SSRF protection: cannot resolve hostname {hostname!r}: {exc}') from exc
+
+    if not addrinfos:
+        raise SSRFError(f'SSRF protection: hostname {hostname!r} resolved to no addresses.')
+
+    resolved_ips: List[str] = []
+    for family, _type, _proto, _canonname, sockaddr in addrinfos:
```

Ruff (0.15.7)

[warning] Loop control variable `family` not used within loop body

Rename unused `family` to `_family`

(B007)